lib/install/management/store_files.go

```diff
@@ -158,26 +158,23 @@ func (d *Dispatcher) isVSAN(ds *object.Datastore) bool {
 func (d *Dispatcher) deleteFilesIteratively(m *object.DatastoreFileManager, ds *object.Datastore, dsPath string) error {
 	defer trace.End(trace.Begin(dsPath, d.op))
 
```

Comment:
sorry for the whiplash Angie - given this isn't a direct revert if addressing anchal's comments, could we add a recursive delete _attempt_ as the first thing, then fail over into the current path.
So adding the following at the top:
```go
// don't check errors given detail process of any remaining files will occur after
d.deleteVMFSFiles(m, ds, dsPath)
```

Reply:
that's alright. unnecessary recursive delete doesn't sound very neat tbh....
it seems like ESX and older versions of VC doesn't have the issue but vSAN and new VC version do have that, in most cases it has to go thru iterative delete

if we add `d.deleteVMFSFiles(m, ds, dsPath)` at the top, then it would probably print failure log messages complaining not being able to delete non-empty datastore folders. Since we ignore the error coming out of this, and recursively clean stuff up later, if the folder did get cleaned up in the end, looking at the log it would still feel like that there's a failure deleting the folder.
sorry about the messy description, but just wondering will it cause bad user experience?

Reply:
It looks like this from a test run I did:
```
INFO[0003] Removing image stores 
DEBU[0004] Failed to delete "[datastore1] vch-test-4/VIC": Cannot delete file [datastore1] vch-test-4/VIC 
DEBU[0013] Folder "[datastore1] vch-test-4/kvStores" is not found 
DEBU[0013] Image store parent directory not empty, leaving in place.
......
DEBU[0016] Folder "[datastore1] vch-test-4" is not found 
INFO[0016] Removing Resource Pool "vch-test-4" 
```
The later `Folder "[datastore1] vch-test-4" is not found` is from `vm.Unregsiter`, it shows that the folder is already been deleted
but still we see ` Failed to delete "[datastore1] vch-test-4/VIC": Cannot delete file [datastore1] vch-test-4/VIC` on the top, even tho it's actually deleted successfully right after

Reply:
I'm okay with them showing up as debug messages - we could add an additional debug message saying "Attempting recursive delete" then if that's not clean a message saying "Cleaning up recusive deletion failures via iterative delete" or similar.

We may want to return a different error type rather than re-use NotFound if it's causing misleading output.

Reply:
Got it. Just experimented and here's the log output:
```
DEBU[0007] No disks found attached to VM                
INFO[0007] Removing image stores                        
DEBU[0009] Failed to delete "[datastore1] vch-2/VIC": Cannot delete file [datastore1] vch-2/VIC 
DEBU[0009] Attempt to delete top level folder "[datastore1] vch-2/VIC" failed. Removing the children files instead. 
DEBU[0027] Image store parent directory not empty, leaving in place. 
INFO[0027] Removing volume stores 
```
Does this look okay? I added a log saying `Attempt to delete top level folder failed. Removing the children files instead`